sender.go

```diff
@@ -41,10 +41,6 @@ func (s *Sender) MaxMessageSize() uint64 {
 // additional messages can be sent while the current goroutine is waiting
 // for the confirmation.
 func (s *Sender) Send(ctx context.Context, msg *Message) error {
-	if err := s.link.Check(); err != nil {
-		return err
-	}
-
 	done, err := s.send(ctx, msg)
```

Comment:
Looking through here (just browsing the code) it seems like it's not harmful to go through the .send() path, but I can't see a reason to do it, which is what the Check() call was helping with.

Do we need to remove it?

Reply:
send will properly exit if the link/session/conn has been closed, or the link has been detached.  Or are you suggesting adding an early-exit if link.Detached is closed?

Reply:
Added a check to exit early from `Send` when the link is detached.